### PR TITLE
Retry + backoff on GCS timeouts

### PR DESCRIPTION
**Pull request by kjetilh33**
Add GCS timeouts to the collection of exceptions that will be retried

### CODE REVIEW

**Review by pavelzubarev — COMMENTED** [review]
How is ClientRequestException connected to RETRIABLE_EXCEPTIONS?

File: src/main/java/com/cognite/client/Files.java [hunk 1]
```diff
@@ -955,18 +955,8 @@ public Aggregate aggregate(Request requestParameters) throws Exception {
...
-                if (e instanceof FileBinaryRequestExecutor.ClientRequestException) {
-                    requestException = (FileBinaryRequestExecutor.ClientRequestException) e;
-                } else if (e instanceof CompletionException) {
-                    // Unwrap the completion exception to see if the underlying cause is a ClientRequestException
-                    requestException = ((CompletionException) e).getCause() instanceof FileBinaryRequestExecutor.ClientRequestException ?
-                            ((FileBinaryRequestExecutor.ClientRequestException) ((CompletionException) e).getCause()) : null;
-                }
-
-                if (null != requestException) {
+            } catch (CompletionException e) {
```
Comment (pavelzubarev) [inline src/main/java/com/cognite/client/Files.java:958]: that change is reducing on handling ClientRequestException... Is that intended?

**Comment by kjetilh33** [comment]
> How is ClientRequestException connected to RETRIABLE_EXCEPTIONS?

It is different. The ClientRequestException is a special handling of 400 responses (basically when a URL has expired). We can't do request retries on that one (since the URL is expired), so we have to handle it on the batch-level by re-trying the entire two-step process (get the URL + call the URL). This handles the special case of downloading a binary from a URL that has time expiration.

RETRIABLE_EXCEPTIONS will lead to requests being retried as is. This is the default retry + backoff mechanism.

File: src/main/java/com/cognite/client/Files.java (hunk 1 above)
Comment (kjetilh33) [reply src/main/java/com/cognite/client/Files.java:958]: Yes, it was a cleanup of the previous code that was a bit too unreadable. 

Basically, this code should just handle the ClientRequestException.

**Review by pavelzubarev — APPROVED** [review]
Ok. I see

**Pull request merged**